File: 94. Binary_Tree_Inorder_Traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    List<Integer> ans = new ArrayList<Integer>();
    
    public void traverse(TreeNode cur_node){
        
        if(cur_node.left != null)
            traverse(cur_node.left);
        
        ans.add(cur_node.val);
        
        if(cur_node.right != null)
            traverse(cur_node.right);
        
        return;
    }
    
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root != null)
            traverse(root);
        return ans;    
    }
}

File: 94. Binary_Tree_Inorder_Traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> ans;
    void DFS(TreeNode* cur_node){
        if(cur_node != nullptr){
            if(cur_node->left != nullptr)
                DFS(cur_node->left);
            ans.push_back(cur_node->val);
            if(cur_node->right != nullptr)
                DFS(cur_node->right);
        }
        
    }
    vector<int> inorderTraversal(TreeNode* root) {
        DFS(root);
        return ans;
    }
};